lib/bolt/outputter/human.rb

```diff
@@ -1,27 +1,80 @@
 module Bolt
   class Outputter
     class Human < Bolt::Outputter
+      COLORS = { red: "31",
+                 green: "32",
+                 yellow: "33" }.freeze
+
       def print_head; end
 
+      def colorize(color, string)
+        if @stream.isatty
+          "\033[#{COLORS[color]}m#{string}\033[0m"
+        else
+          string
+        end
+      end
+
+      def indent(indent, string)
+        indent = ' ' * indent
+        string.gsub(/^/, indent.to_s)
+      end
+
+      def remove_trail(string)
+        string.sub(/\s\z/, '')
+      end
+
+      def print_event(node, event)
+        case event[:type]
+        when :node_start
+          print_start(node)
+        when :node_result
+          print_result(node, event[:result])
+        end
+      end
+
+      def print_start(node)
+        @stream.puts(colorize(:green, "Started on #{node.host}..."))
+      end
+
       def print_result(node, result)
-        color = result.success? ? "\033[32m" : "\033[31m"
-        @stream.print color if @stream.isatty
-        @stream.puts "#{node.host}:"
-        @stream.print "\033[0m" if @stream.isatty
-        @stream.puts
-        @stream.puts result.message
-        @stream.puts
+        if result.success?
+          @stream.puts(colorize(:green, "Finished on #{node.host}:"))
+        else
+          @stream.puts(colorize(:red, "Failed on #{node.host}:"))
+        end
+
+        if result.error
+          if result.error['msg']
+            @stream.puts(colorize(:red, remove_trail(indent(2, result.error['msg']))))
+          else
+            @stream.puts(colorize(:red, remove_trail(indent(2, result.error))))
+          end
+        end
+
+        if result.message
+          @stream.puts(remove_trail(indent(2, result.message)))
+        end
+
+        if result.instance_of? Bolt::TaskResult
+          @stream.puts(indent(2, ::JSON.pretty_generate(result.value)))
+        elsif result.instance_of? Bolt::CommandResult
+          @stream.puts(indent(2, "STDOUT:"))
```

Comment:
Should thees be printed even if stdout or stderr are empty?

Reply:
I think it's useful to at least have the stdout key printed to make it clear it was empty.